packages/next/src/build/entries.ts

```diff
@@ -580,16 +604,31 @@ export async function createEntrypoints(
               appPaths: matchedAppPaths,
               pageExtensions,
               assetPrefix: config.assetPrefix,
-              nextConfigOutput: config.output,
               // This isn't used with edge as it needs to be set on the entry module, which will be the `edgeServerEntry` instead.
               // Still passing it here for consistency.
               preferredRegion: staticInfo.preferredRegion,
+              config,
+            }).import
+          } else if (pagesType === 'pages' && !isAPIRoute(page)) {
+            appDirLoader = getNextRouteModuleEntry({
```

Comment:
Should this variable be renamed to `moduleLoader` or something similar? Seems it's no longer just the App Router loader.